### ISSUE TITLE
[ToS] Comply with the Expires header

### ISSUE THREAD

**Issue by papjul** [WangDaYeeeeee/GeometricWeather#244]
Hi,

Many weather providers request that developers should/must comply with the Expires HTTP header.

The Expires header contains the date/time after which data will be updated on server side. Until then, it is useless (and harmful for the bandwidth) to refresh.

As per AccuWeather guidelines:
```
Refresh information from the AccuWeather API for your device based upon the cache expires time in the response headers

Example:
   Response Headers
    Cache-Control: public
    Content-Encoding: gzip
    Content-Type: application/json; charset=utf-8
    Date: Wed, 29 Aug 2012 14:55:33 GMT
    Expires: Thu, 30 Aug 2012 14:56:34 GMT
    Server: Microsoft-IIS/7.5
    Server: Microsoft-IIS/7.0
    Transfer-Encoding: chunked
    Vary: Accept-Encoding
    X-AspNet-Version: 4.0.30319
    X-Powered-By: ASP.NET

In this example, do not refresh until: Thu, 30 Aug 2012 14:56:34 GMT
```

This is also a prerequisite for #242 

Would it be possible to make sure data is not refreshed until that date?
This would apply to:
- Manual refreshes (if some users force refreshing before the Expires date)
- Background refreshes (if configured to refresh every 1 hour, if the Expires is in 1 h 30, then you should not refresh before 1 h 30 (or 2 h if you want to stay every regular 1 hour)

This would also save up some precious API calls from useless refreshes, as seen in #223.

**Comment by rthery** [comment]
Hum, good http client library usually handles that for you. The request will be cached until the Expires value, and any identical requests issued will just return the cached response immediately.

I'm not familiar with Android API, but this looks like this could do the job ? (2min googlefu, may be wrong :D)
https://developer.android.com/reference/android/net/http/HttpResponseCache

**Comment by papjul** [comment]
We use the Retrofit dependency to ease API handling. According to my HTTP interceptor, it makes a new request everytime.

**Comment by rthery** [comment]
I see, OkHttpClient (which is used by Retrofit) supports caching, and from what I can see here the cache is not enabled:
https://github.com/WangDaYeeeeee/GeometricWeather/blob/master/app/src/main/java/wangdaye/com/geometricweather/common/retrofit/TLSCompactHelper.java#L90

I can do some tests locally and issue a PR

**Comment by papjul** [comment]
Fixed in #440
